vizro-ai/examples/dashboard_ui/components.py

```diff
@@ -0,0 +1,216 @@
+            ],
+            className="code-clipboard-container",
+        )
+
+
+class MyDropdown(vm.Dropdown):
+    """Custom dropdown component."""
+
+    type: Literal["my_dropdown"] = "my_dropdown"
+
+    def build(self):
+        """Returns custom dropdown component that cannot be cleared."""
+        dropdown_build_obj = super().build()
+        dropdown_build_obj.id = f"{self.id}_outer_div"
+        dropdown_build_obj.children[1].clearable = False
+
+        return dropdown_build_obj
+
+
+class OffCanvas(vm.VizroBaseModel):
+    """OffCanvas component for settings."""
+
+    type: Literal["offcanvas"] = "offcanvas"
+    options: List[str]
+    value: str
+
+    def build(self):
+        """Returns the off canvas component for settings."""
+        input_groups = html.Div(
+            [
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText("API Key"),
+                        dbc.Input(placeholder="API key", type="password", id=f"{self.id}-api-key"),
+                        html.Div(
+                            dbc.Checklist(
+                                id=f"{self.id}-api-key-toggle",
+                                options=[{"label": "", "value": False}],
+                                switch=True,
+                                inline=True,
+                            ),
+                            id="toggle-div-api-key",
+                        ),
+                    ],
+                    className="mb-3",
+                ),
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText("API base"),
+                        dbc.Input(placeholder="(optional) API base", type="password", id=f"{self.id}-api-base"),
+                        html.Div(
+                            dbc.Checklist(
+                                id=f"{self.id}-api-base-toggle",
+                                options=[{"label": "", "value": False}],
+                                switch=True,
+                                inline=True,
+                            ),
+                            id="toggle-div-api-base",
+                        ),
+                    ],
+                    className="mb-3",
+                ),
+                dbc.InputGroup(
+                    [
+                        dbc.InputGroupText("Choose your vendor"),
+                        dbc.Select(options=self.options, value=self.value, id=f"{self.id}-dropdown"),
+                    ],
+                    className="mb-3",
+                ),
+            ],
+            className="mb-3",
+        )
+
+        offcanvas = dbc.Offcanvas(
+            id=self.id,
+            children=[
+                html.Div(
+                    children=[
+                        input_groups,
+                    ]
+                )
+            ],
+            title="Settings",
+            is_open=True,
+        )
+        return offcanvas
+
+
+class MyPage(vm.Page):
```

Comment:
Why do we do this?

Reply:
We needed to overwrite Page pre_build method, to be able to use `Input("on_page_load_action_trigger_vizro_ai_plot_page", "data")` as input in the callback to retrieve the latest outputs of vizro-ai when navigating to and from another page. 
But now, since we have only one page, this can be removed.